src/ext/oblivc/copy.c

```diff
@@ -260,18 +262,42 @@ void ocFromShared_impl(ProtocolDesc* pd,
   // Let's go with malloc for now, we change that later.
   struct OcShareContext* ctx = protoShareCtx(pd);
   YaoProtocolDesc* ypd = pd->extra;
+#ifdef USE_PLAIN_OT
+  OTsender* sender;
+  OTrecver* recver;
+  if (ctx == NULL) {
+    sender = &ypd->sender;
+    recver = &ypd->recver;
+  } else {
+    sender = &ctx->sender;
```

Comment:
Yes, the idea here was to allow this code to do something sensible in the case that ocShareInit hasn't been called yet (in which case ctx will be null and everything in the else clause will be a dangling pointer). Why might we want that to be the case? I'm using ocFromShared/ocToShared in a library, and I don't know whether the application code has called ocSharedInit(). As designed, it can't be called twice (or so it seems to me, anyway). The alternative is to add a check so that ocSharedInit() becomes a no-op if it's already been run. I'm perfectly OK with that solution if you'd prefer it, although I'm not sure that it's really "proper" to split the protocol in a library if the application hasn't done so. Thoughts?